src/azure-cli/azure/cli/command_modules/acr/connected_registry.py

```diff
@@ -440,3 +473,144 @@ def _get_install_info(cmd,
+    else:
+        action_txt = "Adding"
+        add_actions_set = set(parse_scope_map_actions(repository_actions_list, gateway_actions_list))
+        remove_actions_set = set()
+
+    while parent_id and not parent_id.isspace():
+        ancestor = family_tree[parent_id]["connectedRegistry"]
+        msg = "{} '{}' gateway permissions{} to connected registry '{}' sync scope map.".format(
+            action_txt, gateway, repo_msg, ancestor.name)
+        _update_repo_permissions(cmd, resource_group_name, registry_name,
+                                 ancestor, add_actions_set, remove_actions_set, msg=msg)
+        parent_id = ancestor.parent.id
+
+
+# region connected-registry repo update
+def _update_repo_permissions(cmd,
+                             resource_group_name,
+                             registry_name,
+                             connected_registry,
+                             add_actions_set,
+                             remove_actions_set,
+                             msg=None,
+                             description=None):
+    scope_map_client = cf_acr_scope_maps(cmd.cli_ctx)
+    sync_token = get_token_from_id(cmd, connected_registry.parent.sync_properties.token_id)
+    sync_scope_map = get_scope_map_from_id(cmd, sync_token.scope_map_id)
+    sync_scope_map_name = sync_scope_map.name
+    current_actions_set = set(sync_scope_map.actions)
+    final_actions_set = current_actions_set.union(add_actions_set).difference(remove_actions_set)
+    if final_actions_set == current_actions_set:
+        return None
+    current_actions = list(final_actions_set)
+    logger.warning(msg)
+    return scope_map_client.update(
+        resource_group_name,
+        registry_name,
+        sync_scope_map_name,
+        description,
+        current_actions
+    )
+
+
+def _get_scope_map_actions_set(repos, actions):
+    for i, repo_name in enumerate(repos):
+        repos[i] = [repo_name] + actions
+    return set(parse_scope_map_actions(repos))
+
+
+def acr_connected_registry_repo(cmd,
+                                client,
+                                connected_registry_name,
+                                registry_name,
+                                add_repos=None,
+                                remove_repos=None,
+                                resource_group_name=None):
+    if not (add_repos or remove_repos):
+        raise CLIError('No repository permissions to update.')
+    _, resource_group_name = validate_managed_registry(
+        cmd, registry_name, resource_group_name)
+
+    add_repos_set = set(add_repos) if add_repos is not None else set()
+    remove_repos_set = set(remove_repos) if remove_repos is not None else set()
+    duplicate_repos = set.intersection(add_repos_set, remove_repos_set)
+    if duplicate_repos:
+        errors = sorted(map(lambda action: action[action.rfind('/') + 1:], duplicate_repos))
```

Comment:
Do you have an example of errors?

Reply:
do you mean something like --add repo1 --delete repo1? Something like that? This is pretty much the same message we use in other places we have a similar check